services/src/datasets/external/sentinel_s2_l2a_cogs.rs

```diff
@@ -0,0 +1,640 @@
+            }
+        }
+        debug!("number of generated loading infos: {}", parts.len());
+
+        Ok(GdalLoadingInfo {
+            info: GdalLoadingInfoPartIterator::Static {
+                parts: parts.into_iter(),
+            },
+        })
+    }
+
+    fn create_loading_info_part(
+        &self,
+        time_interval: TimeInterval,
+        asset: &StacAsset,
+    ) -> Result<GdalLoadingInfoPart> {
+        Ok(GdalLoadingInfoPart {
+            time: time_interval,
+            params: GdalDatasetParameters {
+                file_path: PathBuf::from(format!("/vsicurl/{}", asset.href)),
+                rasterband_channel: 1,
+                geo_transform: GeoTransform::from(
+                    asset
+                        .gdal_geotransform()
+                        .ok_or(error::Error::StacInvalidGeoTransform)?,
+                ),
+                bbox: asset
+                    .native_bbox()
+                    .ok_or(error::Error::StacInvalidBbox)?
+                    .into(),
+                file_not_found_handling: geoengine_operators::source::FileNotFoundHandling::NoData,
+                no_data_value: self.band.no_data_value,
+                properties_mapping: None,
+            },
+        })
+    }
+
+    fn request_params(&self, query: QueryRectangle) -> Result<Vec<(String, String)>> {
+        let (t_start, t_end) = Self::time_range_request(&query.time_interval)?;
+
+        // request all features in zone in order to be able to determine the temporal validity of individual tile
+        let bbox =
+            SpatialReference::new(SpatialReferenceAuthority::Epsg, self.zone.epsg).area_of_use()?;
+
+        Ok(vec![
+            (
+                "collections[]".to_owned(),
+                "sentinel-s2-l2a-cogs".to_owned(),
+            ),
+            (
+                "bbox".to_owned(),
+                format!(
+                    "[{},{},{},{}]", // array-brackets are not used in standard but required here for unknkown reason
+                    bbox.lower_left().x,
+                    bbox.lower_left().y,
+                    bbox.upper_right().x,
+                    bbox.upper_right().y
+                ),
+            ), // TODO: order coordinates depending on projection
+            (
+                "datetime".to_owned(),
+                format!("{}/{}", t_start.to_rfc3339(), t_end.to_rfc3339()),
+            ),
+            ("limit".to_owned(), "500".to_owned()),
+        ])
+    }
+
+    async fn load_all_features<T: Serialize + ?Sized>(
+        &self,
+        params: &T,
+    ) -> Result<Vec<StacFeature>> {
+        let mut features = vec![];
+
+        let mut collection = self.load_collection(params, 1).await?;
+        features.append(&mut collection.features);
+
+        let num_pages =
+            (collection.context.matched as f64 / collection.context.limit as f64).ceil() as u32;
+
+        for page in 2..=num_pages {
```

Comment:
what happens if num_pages is 1?

Reply:
then no further pages have to be downloaded. the first page is always downloaded